sql: TestDropIndex failed

sql.TestDropIndex [failed](https://tanzanite.cluster.engflow.com/invocation/fa25fbe3-5330-4b4d-9e82-5f36b16f3a0c) on master @ [fbb3446238434b494751af2fe699a4941e43fdd6](https://github.com/cockroachdb/cockroach/commits/fbb3446238434b494751af2fe699a4941e43fdd6):


```
=== RUN   TestDropIndex
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDropIndex7222868
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:490: expected 0 key value pairs, but got 401
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestDropIndex7222868
--- FAIL: TestDropIndex (1.44s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=16</code>
, <code>shard=16</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropIndex.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-33560

sql.TestDropIndex [failed](https://tanzanite.cluster.engflow.com/invocation/31fc60dc-f1fa-4f62-82a6-06c671250d09) on master @ [1baa5c9a075ff0791490092898d610df70998fd3](https://github.com/cockroachdb/cockroach/commits/1baa5c9a075ff0791490092898d610df70998fd3):


```
=== RUN   TestDropIndex
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestDropIndex3463169653
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:490: expected 0 key value pairs, but got 401
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestDropIndex3463169653
--- FAIL: TestDropIndex (4.08s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=24</code>
, <code>shard=16</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropIndex.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropIndex [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12731859?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12731859?buildTab=artifacts#/) on master @ [8d026f9782eb929327c8a231fd8a3f979e621931](https://github.com/cockroachdb/cockroach/commits/8d026f9782eb929327c8a231fd8a3f979e621931):


```
=== RUN   TestDropIndex
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b95a0f04eb6b274a388bfa7f20f6c7c4/logTestDropIndex33141777
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:490: expected 0 key value pairs, but got 401
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b95a0f04eb6b274a388bfa7f20f6c7c4/logTestDropIndex33141777
--- FAIL: TestDropIndex (2.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropIndex.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/drop_test.go
+++ b/pkg/sql/drop_test.go
@@ -417,6 +417,14 @@ func TestDropIndex(t *testing.T) {
 	s, sqlDB, kvDB := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.Background())
 
+	_, err := sqlDB.Exec(`SET CLUSTER SETTING sql.gc_job.wait_for_gc.interval = '1s';`)
+	require.NoError(t, err)
+
+	// Refresh protected timestamp cache immediately to make MVCC GC queue to
+	// process GC immediately.
+	_, err = sqlDB.Exec(`SET CLUSTER SETTING kv.protectedts.poll_interval = '1s';`)
+	require.NoError(t, err)
+
 	// Disable strict GC TTL enforcement because we're going to shove a zero-value
 	// TTL into the system with AddImmediateGCZoneConfig.
 	defer sqltestutils.DisableGCTTLStrictEnforcement(t, sqlDB)()
@@ -426,6 +434,9 @@ func TestDropIndex(t *testing.T) {
 		t.Fatal(err)
 	}
 	tableDesc := desctestutils.TestingGetPublicTableDescriptor(kvDB, keys.SystemSQLCodec, "t", "kv")
+	if _, err := sqltestutils.AddImmediateGCZoneConfig(sqlDB, tableDesc.GetID()); err != nil {
+		t.Fatal(err)
+	}
 	tests.CheckKeyCount(t, kvDB, tableDesc.TableSpan(keys.SystemSQLCodec), 3*numRows)
 	idx, err := catalog.MustFindIndexByName(tableDesc, "foo")
 	if err != nil {

--- a/pkg/sql/schemachanger/scexec/gc_jobs.go
+++ b/pkg/sql/schemachanger/scexec/gc_jobs.go
@@ -233,7 +233,6 @@ func createGCJobRecord(
 			descriptorIDs = append(descriptorIDs, details.ParentID)
 		}
 	}
-	runningStatus := jobs.RunningStatus("waiting for MVCC GC")
 	return jobs.Record{
 		JobID:         id,
 		Description:   "GC for " + description,
@@ -241,7 +240,6 @@ func createGCJobRecord(
 		DescriptorIDs: descriptorIDs,
 		Details:       details,
 		Progress:      jobspb.SchemaChangeGCProgress{},
-		RunningStatus: runningStatus,
 		NonCancelable: true,
 	}
 }

